src/Analyzers/CSharp/Analyzers/UseAutoProperty/CSharpUseAutoPropertyAnalyzer.cs

```diff
@@ -95,33 +96,63 @@ SyntaxKind.RecordDeclaration or
                         if (addressOfExpression.Kind() == SyntaxKind.AddressOfExpression)
                             AddIneligibleFields(semanticModel, addressOfExpression.Operand, ineligibleFields, cancellationToken);
                     }
+
+                    foreach (var memberAccess in typeDeclaration.DescendantNodesAndSelf().OfType<MemberAccessExpressionSyntax>())
+                        AddIneligibleFieldsIfAccessedOffNotDefinitelyAssignedValue(semanticModel, memberAccess, ineligibleFields, cancellationToken);
                 }
             }
         }
 
+        private static void AddIneligibleFieldsIfAccessedOffNotDefinitelyAssignedValue(
+            SemanticModel semanticModel, MemberAccessExpressionSyntax memberAccess, HashSet<IFieldSymbol> ineligibleFields, CancellationToken cancellationToken)
+        {
+            // `c.x = ...` can't be converted to `c.X = ...` if `c` is a struct and isn't definitely assigned as that point.
+
+            // only care about writes.  if this was a read, then it must be def assigned and thus is safe to convert to a prop.
+            if (!memberAccess.IsOnlyWrittenTo())
+                return;
+
+            var symbolInfo = semanticModel.GetSymbolInfo(memberAccess, cancellationToken);
+            if (symbolInfo.GetAnySymbol() is not IFieldSymbol)
+                return;
+
+            // this only matters for a struct.  They can be declared unassigned and have their fields directly written into.
+            var exprType = semanticModel.GetTypeInfo(memberAccess.Expression, cancellationToken).Type;
+            if (exprType?.TypeKind != TypeKind.Struct)
```

Comment:
❔ Can we avoid the call to `GetTypeInfo` by using `IFieldSymbol.DeclaringType` from the previous check?

Reply:
yup.  that makes sense.  will fix.